feat: add `math/base/special/fibonaccif`

anandkaranubc opened the pull request:
Progresses #649

## Description

> What is the purpose of this pull request?

This pull request:

- Adds the JS and C implementation of `math/base/special/fibonaccif`.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #649

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

anandkaranubc committed:
feat: add complete implementation

anandkaranubc committed:
Merge branch 'stdlib-js:develop' into feat/fibonaccif

anandkaranubc commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

anandkaranubc committed:
chore: fix examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: use correct data types

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/fibonaccif/lib/main.js:
```suggestion
* Computes the nth Fibonacci number and returns the result as a single-precision floating-point number.
```

This one seems more clear.

kgryte commented on lib/node_modules/@stdlib/math/base/special/fibonaccif/lib/main.js:
"Computes the nth Fibonacci number as a single-precision floating-point number."?

anandkaranubc committed:
docs: fix the function description

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
test: fix test messages

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
Merge branch 'stdlib-js:develop' into feat/fibonaccif

anandkaranubc committed:
chore: re-enable lint rule

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/math/base/special/fibonaccif/benchmark/c/native/benchmark.c:
Why are we doing so few iterations here?

Ref: https://github.com/stdlib-js/stdlib/blob/develop/tools/snippets/benchmark/c/benchmark.c#L26

kgryte commented on lib/node_modules/@stdlib/math/base/special/fibonaccif/benchmark/c/benchmark.c:
Why are we doing so few iterations here?

Ref: https://github.com/stdlib-js/stdlib/blob/develop/tools/snippets/benchmark/c/benchmark.c#L26

kgryte commented on lib/node_modules/@stdlib/math/base/special/fibonaccif/benchmark/c/native/benchmark.c:
Based on Git blame, this was me: https://github.com/stdlib-js/stdlib/blame/develop/lib/node_modules/%40stdlib/math/base/special/fibonacci/benchmark/c/benchmark.c

Maybe it has to due to stack limits.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/fibonaccif/benchmark/c/native/benchmark.c:
No reason. I just got tripped up when referencing `fibonacci`. ~I'll change that.~

kgryte reviewed (approved):
LGTM
